frontend/packages/topology/integration-tests/support/step-definitions/topology/chart-view.ts

```diff
@@ -0,0 +1,128 @@
+import { When, Then, Given } from 'cypress-cucumber-preprocessor/steps';
+import { pageTitle } from '@console/dev-console/integration-tests/support/constants';
+import { devNavigationMenu } from '@console/dev-console/integration-tests/support/constants/global';
+import { createGitWorkload } from '@console/dev-console/integration-tests/support/pages';
+import { app, navigateTo } from '@console/dev-console/integration-tests/support/pages/app';
+import { topologyPO, typeOfWorkload } from '../../page-objects/topology-po';
+import { topologyHelper } from '../../pages/topology';
+
+When('user navigates to Topology page', () => {
+  navigateTo(devNavigationMenu.Topology);
+});
+
+Then('user sees Topology page with message {string}', (message: string) => {
+  cy.get(topologyPO.emptyText).should('have.text', message);
+});
+
+Given('user is at the Topology page', () => {
+  navigateTo(devNavigationMenu.Topology);
+});
+
+When('user clicks on {string} link in the topology page', (addLink) => {
+  cy.get(topologyPO.graph.addLink)
+    .should('be.visible')
+    .should('have.text', addLink);
+  cy.get(topologyPO.graph.addLink).click();
+});
+
+Then('user will be redirected to Add page', () => {
+  cy.url().should('include', 'add');
+  app.waitForLoad();
+  cy.contains(pageTitle.Add).should('be.visible');
+});
+
+Given('user is at the Topology page', () => {
+  navigateTo(devNavigationMenu.Topology);
+});
+
+When('user clicks on {string} link in the empty topology page', (build: string) => {
+  cy.get(topologyPO.graph.startBuildingNewApp)
+    .should('be.visible')
+    .click();
```

Comment:
Little bit confusing that you don't use `build: string` here.

Reply:
Updated